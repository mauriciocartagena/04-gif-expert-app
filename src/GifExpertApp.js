import React,{ useState } from 'react';
//snipet para crear todo de una rafc 

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];

    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    const handleAdd = () =>{
       
        //Es una forma de agregar al array nuevos valores
        setCategories( [ ...categories, 'Hunter' ] );


        //Es una forma de agregar al array nuevos valores pero inversa ahora
        // setCategories( [ 'Hunter' , ...categories ] );

        //Otra forma de agregar
        // setCategories( cats => [ ...cats,'Hunter' ]);

    }


    return (

        <React.Fragment>
            <h2>GifExpertApp</h2>
            <hr />

            <button onClick={ handleAdd }>Agregar</button>


            <ol>
                {
                    categories.map(( category ) => {
                    return <li key={ category }> { category }</li>;
                    })
                }
            </ol>


        </React.Fragment>
    )

};

export default GifExpertApp;