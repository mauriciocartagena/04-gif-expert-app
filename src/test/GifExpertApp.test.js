import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas de <GifExpertApp />', () => {

    test('Debe de mostrar correctamente el <GifExpertApp />', () => {

        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();
    });
    test('Debe de mostrar una lista de categorias', () => {

        const categories = ['One punch','Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrif').length ).toBe( categories.length );
         
    });
    

});
