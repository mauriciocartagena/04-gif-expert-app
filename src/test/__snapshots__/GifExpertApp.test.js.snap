// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp /> Debe de mostrar correctamente el <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas de <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrif
      category="One punch"
      key="One punch"
    />
    <GifGrif
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
